### PR TITLE
test: ProcessProposal with invalid namespace in index wrapped PFB tx

### PR DESCRIPTION
**Pull request by rootulp**
Closes https://github.com/celestiaorg/celestia-app/issues/1558

### CODE REVIEW

File: app/test/process_proposal_test.go
Comment (SweeXordious) [inline app/test/process_proposal_test.go:222]: [Question]
Is the invalid namespace == namespace version 1?

**Review by SweeXordious — COMMENTED** [review]
LGTM :+1: 

Would defer to other reviewers for final approval

File: app/test/process_proposal_test.go
Comment (rootulp) [reply app/test/process_proposal_test.go:222]: yep! Specifically https://github.com/rootulp/celestia-app/blob/ceeb7f45bef4d30ffc94760a48c78452505b4d08/testutil/blobfactory/payforblob_factory.go#L408

File: testutil/blobfactory/payforblob_factory.go
Comment (evan-forbes) [inline testutil/blobfactory/payforblob_factory.go:319]: [blocking question]
wait, why was this done? won't this result in not using the caller specified namespaces?

File: testutil/blobfactory/payforblob_factory.go
Comment (evan-forbes) [inline testutil/blobfactory/payforblob_factory.go:401]: ?
```suggestion
	blob := ManyRandBlobs(t, 1)[0]
```

File: testutil/blobfactory/payforblob_factory.go
Comment (evan-forbes) [inline testutil/blobfactory/payforblob_factory.go:381]: optional naming comment

maybe `IndexWrappedTxWithInvalidNamespace`

**Review by evan-forbes — COMMENTED** [review]
one blocking question, otherwise LGTM

File: testutil/blobfactory/payforblob_factory.go
Comment (rootulp) [reply testutil/blobfactory/payforblob_factory.go:401]: `100` is the blob size

https://github.com/rootulp/celestia-app/blob/93f74344b7394173d264151628b5fe076c57c9a0/testutil/blobfactory/payforblob_factory.go#L288

https://github.com/rootulp/celestia-app/blob/93f74344b7394173d264151628b5fe076c57c9a0/testutil/blobfactory/payforblob_factory.go#L300

File: testutil/blobfactory/payforblob_factory.go
Comment (rootulp) [reply testutil/blobfactory/payforblob_factory.go]: resolved by 02443101148451ec134880ea295c06c66d615569

File: testutil/blobfactory/payforblob_factory.go
Comment (rootulp) [reply testutil/blobfactory/payforblob_factory.go]: I was trying to simplify this code but on second look, the change wasn't needed and I don't think it's correct. The fact that no tests broke after this change makes me think that nested blob test data isn't really verified but I'll revert.

Comment (rootulp) [reply testutil/blobfactory/payforblob_factory.go]: resolved by ae01b98416c38a218b0de5d37cebb8c553a6001c

**Comment by rootulp** [comment]
@SweeXordious FWIW I'll need two approvals to merge this so if you don't feel comfortable explicitly approving, I can add another reviewer.

File: testutil/blobfactory/payforblob_factory.go
Comment (evan-forbes) [reply testutil/blobfactory/payforblob_factory.go:403]: lol :face_in_clouds:

File: testutil/blobfactory/payforblob_factory.go
Comment (evan-forbes) [reply testutil/blobfactory/payforblob_factory.go]: > The fact that no tests broke after this change makes me think that nested blob test data isn't really verified

this just overrides the callers namespace, so unless we're checking for the caller's namespace, I wouldn't expect it to be detected

**Review by evan-forbes — APPROVED** [review]

**Review by SweeXordious — APPROVED** [review]
Oh no sorry, i didnt know that was activated. Thanks for pinign

**Pull request merged**
